nautobot/extras/views.py

```diff
@@ -455,6 +458,82 @@ class CustomFieldBulkDeleteView(generic.BulkDeleteView):
     queryset = CustomField.objects.all()
     table = tables.CustomFieldTable
 
+    def construct_custom_field_delete_tasks(self, queryset):
+        """
+        Helper method to construct a list of celery tasks to execute when bulk deleting custom fields.
+        """
+        tasks = [
+            delete_custom_field_data.si(obj.name, set(obj.content_types.values_list("pk", flat=True)))
+            for obj in queryset
+        ]
+        return tasks
+
+    # Overloading post() method for CustomFieldBulkDeleteView
+    def post(self, request, **kwargs):
+        logger = logging.getLogger("nautobot.views.BulkDeleteView")
+        model = self.queryset.model
+
+        # Are we deleting *all* objects in the queryset or just a selected subset?
+        if request.POST.get("_all"):
+            if self.filterset is not None:
+                pk_list = [obj.pk for obj in self.filterset(request.GET, model.objects.only("pk")).qs]
+            else:
+                pk_list = model.objects.values_list("pk", flat=True)
+        else:
+            pk_list = request.POST.getlist("pk")
+
+        form_cls = self.get_form()
+
+        if "_confirm" in request.POST:
+            form = form_cls(request.POST)
+            if form.is_valid():
+                logger.debug("Form validation was successful")
+
+                # Delete objects
+                queryset = self.queryset.filter(pk__in=pk_list)
+                try:
+                    tasks = self.construct_custom_field_delete_tasks(queryset=queryset)
+                    chain(*tasks).apply_async()
+                    deleted_count = queryset.delete()[1][model._meta.label]
```

Comment:
Is the `queryset.delete()` count dependent upon the the chained tasks completing? Are there any exceptions that may bubble up that would need to be caught and may be better left out of this `try` block here?

Also can you please replace the use of array slicing here and instead unpack, e.g. 

```
    _, deleted_info = queryset.delete()
    deleted_count = deleted_info[model._meta.label]
```

It's a tad more verbose but readability counts.